src/detectors/LOWQ2/LowQ2ProtoCluster_factory.cc

```diff
@@ -0,0 +1,132 @@
+// Created by Simon Gardner to do LowQ2 pixel clustering
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#include <JANA/JEvent.h>
+#include <edm4eic/TrackerHit.h>
+#include <edm4eic/RawTrackerHit.h>
+
+#include "LowQ2ProtoCluster_factory.h"
+#include "services/log/Log_service.h"
+#include "extensions/spdlog/SpdlogExtensions.h"
+#include "ROOT/RVec.hxx"
+
+namespace eicrecon {
+
+
+  void LowQ2ProtoCluster_factory::Init() {
+
+    auto app = GetApplication();
+
+    m_log = app->GetService<Log_service>()->logger(m_output_tag);
+
+    m_geoSvc = app->GetService<JDD4hep_service>();
+
+    if(m_readout.empty()){ m_log->error("READOUT IS EMPTY!"); return; }
+
+    try {
+      id_dec = m_geoSvc->detector()->readout(m_readout).idSpec().decoder();
+      if (!m_moduleField.empty()) {
+	module_idx = id_dec->index(m_moduleField);
+	m_log->info("Find module field {}, index = {}", m_moduleField, module_idx);
+      }
+      if (!m_layerField.empty()) {
+	layer_idx = id_dec->index(m_layerField);
+	m_log->info("Find layer field {}, index = {}", m_layerField, layer_idx);
+      }
+      if (!m_xField.empty()) {
+	x_idx = id_dec->index(m_xField);
+	m_log->info("Find layer field {}, index = {}",  m_xField,      x_idx);
+      }
+      if (!m_yField.empty()) {
+	y_idx = id_dec->index(m_yField);
+	m_log->info("Find layer field {}, index = {}", m_yField, y_idx);
+      }
+    } catch (...) {
+      m_log->error("Failed to load ID decoder for {}", m_readout);
+      throw JException("Failed to load ID decoder");
+    }
+
+
+    m_log->info("LowQ2 Tagger Decoding complete...");
+
+  }
+
+
+  void LowQ2ProtoCluster_factory::ChangeRun(const std::shared_ptr<const JEvent> &event) {
+    // Nothing to do here
+  }
+
+  void LowQ2ProtoCluster_factory::Process(const std::shared_ptr<const JEvent> &event) {
```

Comment:
Algorithms like this and the others are supposed to be added to src/algorithms under LGPL3. Unit tests are also much welcome.

Reply:
I'm hoping to split bits of this PR into algorithms to be more in line with the philosophy of EICrecon in a future PR.

Reply:
A reason to split the logic into algorithms ASAP is because we would also like to have someone contribute basic unit tests for this code.